pkg/tsdb/prometheus/framing_test.go

```diff
@@ -4,68 +4,54 @@ import (
 	"bytes"
 	"context"
 	"encoding/json"
-	"fmt"
 	"io"
 	"net/http"
 	"os"
 	"path/filepath"
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/require"
+
 	"github.com/grafana/grafana-plugin-sdk-go/backend"
 	"github.com/grafana/grafana-plugin-sdk-go/experimental"
 	"github.com/grafana/grafana/pkg/infra/tracing"
 	"github.com/prometheus/client_golang/api"
 	apiv1 "github.com/prometheus/client_golang/api/prometheus/v1"
-	"github.com/stretchr/testify/require"
 )
 
 func TestMatrixResponses(t *testing.T) {
-	t.Run("parse a simple matrix response", func(t *testing.T) {
-		err := testScenario("range_simple")
-		require.NoError(t, err)
-	})
-
-	t.Run("parse a simple matrix response with value missing steps", func(t *testing.T) {
-		err := testScenario("range_missing")
-		require.NoError(t, err)
-	})
-
-	t.Run("parse a response with Infinity", func(t *testing.T) {
-		err := testScenario("range_infinity")
-		require.NoError(t, err)
-	})
-
-	t.Run("parse a response with NaN", func(t *testing.T) {
-		err := testScenario("range_nan")
-		require.NoError(t, err)
-	})
-}
+	tt := []struct {
+		name     string
+		filepath string
+	}{
+		{name: "parse a simple matrix response", filepath: "range_sample"},
+		{name: "parse a simple matrix response with value missing steps", filepath: "range_missing"},
+		{name: "parse a response with Infinity", filepath: "range_infinity"},
+		{name: "parse a response with NaN", filepath: "range_nan"},
+	}
+
+	for _, test := range tt {
```

Comment:
These tests don't need a helper function. Instead, they lend themselves to table testing.